### PR TITLE
fix: do not import curl impersonate in http clients init

### PR DESCRIPTION
**Pull request by vdusek**
### Description

- do not import curl impersonate in http clients init
- add extra all for installing everything
- minor improvement of readme & intro docs

### Issues

N/A

### Testing

N/A

### Checklist

- [x] CI passed

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -93,6 +93,7 @@ types-psutil = "~5.9.5.20240205"
 types-python-dateutil = "~2.9.0.20240316"
 
 [tool.poetry.extras]
+all = ["beautifulsoup4", "lxml", "html5lib", "curl-cffi", "playwright"]
```
Comment (janbuchar) [inline pyproject.toml:96]:
```suggestion
all = ["beautifulsoup4", "lxml", "html5lib", "curl-impersonate", "playwright"]
```

Comment (vdusek) [reply pyproject.toml:96]: hmm, that's the thing, the package is called `curl-cffi` but our extra is called according to the client -> `curl-impersonate` ... should we align it to `curl-cffi`?

Comment (vdusek) [reply pyproject.toml:96]: in case of `beautifulsoup4` the extra is also called just `beautifulsoup`

Comment (janbuchar) [reply pyproject.toml:96]: Nah, this is better. Sorry.

File: README.md [hunk 2]
```diff
@@ -75,18 +61,14 @@ Then, install the Playwright dependencies:
...
-pip install 'crawlee[beautifulsoup,playwright]'
-```
-
 Verify that Crawlee is successfully installed:
 
 ```sh
 python -c 'import crawlee; print(crawlee.__version__)'
 ```
 
+For detail installation information see the [Setting up](../introduction/01-setting-up.mdx) documentation page.
```
Comment (janbuchar) [inline README.md:70]:
```suggestion
For detailed installation instructions see the [Setting up](../introduction/01-setting-up.mdx) documentation page.
```

**Review by janbuchar — APPROVED** [review]

**Pull request merged**